src/averageAges.js

```diff
@@ -56,6 +78,37 @@ function calculateWomenAverageAge(people, withChildren) {
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
   // write code here
+  const result = [];
+  let sum = 0;
+
+  if (onlyWithSon) {
+    people.map((elem) => {
+      people.filter(obj => {
+        return (elem.name === obj.mother)
+        && (obj.sex === 'm');
+      }).map(kid => {
+        result.push({
+          mother: elem,
+          kid: kid,
+        });
+      });
+    });
+  } else {
+    people.map((elem) => {
+      people.filter(obj => obj.mother === elem.name).map(kid => {
+        result.push({
+          mother: elem,
+          kid: kid,
+        });
+      });
+    });
+  }
+
+  for (let i = 0; i < result.length; i++) {
```

Comment:
Replace the for loop with a more functional approach, like using .reduce() to calculate the sum.